src/NoThunks/Class.hs

```diff
@@ -506,6 +509,20 @@ deriving via (Maybe a) instance NoThunks a => NoThunks (Monoid.Last a)
 deriving via (f a) instance NoThunks (f a) => NoThunks (Monoid.Alt f a) 
 deriving via (f a) instance NoThunks (f a) => NoThunks (Monoid.Ap f a) 
 
+{-------------------------------------------------------------------------------
+  Solo
+-------------------------------------------------------------------------------}
+
+#if MIN_VERSION_base(4,16,0)
+-- GHC-9.2
+instance NoThunks a => NoThunks (Solo a) where
+    wNoThunks ctx (Solo a) = wNoThunks ("Solo" : ctx) a
+#elif MIN_VERSION_base(4,17,0)
+-- GHC-9.4 and newer
+instance NoThunks a => NoThunks (Solo a) where
+    wNoThunks ctx (MkSolo a) = wNoThunks ("Solo" : ctx) a
+#endif
```

Comment:
This needs to be the other way around (first the `MIN_VERSION_base(4,17,0)` case, then the `MIN_VERSION_base(4,16,0)` case). Currently, the first case is also picked for GHC 9.4 and newer.